src/main/java/com/example/DataService/CurrencyService.java

```diff
@@ -134,5 +139,57 @@ public void updateHistory(){
 	public Collection<Currency> getAllCurrencies() {
 		return cr.findAll();
 	}
+	
+	
+	public BigDecimal currencyExchange(Currency from, Currency to) {
+		if(from == null || to == null) {
+			logger.error("There is no such currency");
+			return new BigDecimal(-1);
+		}
+		
+		return from.getPriceusd().divide(to.getPriceusd(), 16, RoundingMode.CEILING);		
+	}
+	
+	
+	public BigDecimal currencyExchangeIntoRealCurrency(Currency from, String to) {
+		
+		RealCurrencyDTO toCurrency = new RealCurrencyDTO();
+
+		RestTemplate restTemplate = new RestTemplate();
+		ResponseEntity<String> response = restTemplate.getForEntity(exchangeUri, String.class);
```

Comment:
What is this used for? I just see the status code check

Reply:
It's just to check status code in order to see if the url is valid.

Reply:
You should have error handling for every API call. No need to call an endpoint to see if it is valid - that does not tell us anything. We can discuss this together and i will try to explain as much as i can.